hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestZKConnectionRegistry.java

```diff
@@ -125,4 +127,20 @@ public void testNoMetaAvailable() throws InterruptedException {
       }
     }
   }
+
+  /**
+   * Pass discontinuous list of znodes to registry getMetaRegionLocation. Should work fine.
+   * It used to throw ArrayOutOfBoundsException. See HBASE-25280.
+   */
+  @Test
+  public void testDiscontinuousLocations() throws ExecutionException, InterruptedException {
+    Configuration conf = new Configuration(TEST_UTIL.getConfiguration());
+    try (ZKConnectionRegistry registry = new ZKConnectionRegistry(conf)) {
+      CompletableFuture<RegionLocations> cf = new CompletableFuture<>();
+      List<String> znodes = Arrays.asList(
+        new String[] { ZNodePaths.META_ZNODE_PREFIX, ZNodePaths.META_ZNODE_PREFIX + "_003" });
+      registry.getMetaRegionLocation(cf, znodes);
```

Comment:
Better to not expose this method here but write fake znode on zk to simulate discontinuous meta replicas? I do not think it is very difficult to do...

Reply:
Done